### PR TITLE
Prevent transaction rollback on collection.Create ErrExists error

### PR DESCRIPTION
**Pull request by Tryneus**
Reported in https://github.com/pachyderm/pachyderm/pull/5998#discussion_r617015950.

### CODE REVIEW

File: src/internal/collection/postgres_collection_test.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ func TestPostgresCollections(suite *testing.T) {
 			require.NoError(t, listener.Close())
 		})
 
-		testCol := col.NewPostgresCollection(db, listener, &col.TestItem{}, []*col.Index{TestSecondaryIndex}, nil)
```
Comment (Tryneus) [inline src/internal/collection/postgres_collection_test.go:38]: Unrelated - these tests had been broken on the integration branch.

File: src/internal/collection/postgres_collection.go [hunk 2]
```diff
@@ -90,7 +90,7 @@ func (c *postgresCollection) ReadWrite(tx *sqlx.Tx) PostgresReadWriteCollection
 // NewSQLTx starts a transaction on the given DB, passes it to the callback, and
 // finishes the transaction afterwards. If the callback was successful, the
 // transaction is committed. If any errors occur, the transaction is rolled
-// back.  This will reattempt the transaction up to three times.
+// back.  This will reattempt the transaction forever.
```
Comment (acohen4) [inline src/internal/collection/postgres_collection.go:93]: @Tryneus why do we want to re-attempt the transaction forever?

Comment (Tryneus) [reply src/internal/collection/postgres_collection.go:93]: That's actually the existing behavior for `EtcdCollection`, this is just providing the same interface.

**Review by acohen4 — APPROVED** [review]
Looks good!

**Review by PFedak — APPROVED** [review]
LGTM
learning so much about databases lately, I guess an alternative to this would be a SAVEPOINT + ROLLBACK TO, but this seems much better.

**Pull request merged**